openquakeplatform/openquakeplatform/updatures/app.py

```diff
@@ -0,0 +1,931 @@
+import openquakeplatform.updatures.models_descr.maps
+import openquakeplatform.updatures.models_descr.people
+import openquakeplatform.updatures.models_descr.security
+import openquakeplatform.updatures.models_descr.taggit
+import openquakeplatform.updatures.models_descr.tests
+import openquakeplatform.updatures.models_descr.vulnerability
+import pdb
+
+def pdebug(level, s):
+    if updatures.debug_level < level:
+        return
+    print >>sys.stderr, s
+
+
+def rebuild_order():
+    reloop = True
+    first_loop = True
+    models_descr_old = models_descr.copy()
+    models_descr_rest = {}
+    models_order = []
+    while reloop:
+        for model, descr in models_descr_old.iteritems():
+            pdebug(2, "rebuild_order: model %s" % model)
+            if first_loop:
+                if descr.refs == {} and not descr.inher:
+                    models_order.append(model)
+                else:
+                    models_descr_rest[model] = descr
+            else:
+                for ref_field, ref in descr.refs.iteritems():
+                    if isinstance(ref.model, types.FunctionType):
+                        continue
+                    if ref.model not in models_order:
+                        models_descr_rest[model] = descr
+                        break
+                else:
+                    if (not descr.inher) or (descr.inher in models_order):
+                        models_order.append(model)
+                    else:
+                        models_descr_rest[model] = descr
+
+        models_descr_old = models_descr_rest.copy()
+        models_descr_rest = {}
+        if models_descr_old == {}:
+            reloop = False
+        first_loop = False
+
+    return models_order
+
+class ItemsInfo(object):
+
+    def __init__(self, item, pk_natural=False):
+        self.occur = 1
+        if pk_natural:
+            self.maxpk = 'pk_is_natural'
+        else:
+            self.maxpk = item['pk']
+
+    def update(self, item):
+        self.occur += 1
+        if str(self.maxpk) != 'pk_is_natural':
+            self.maxpk = item['pk'] if item['pk'] > self.maxpk else self.maxpk
+
+    def newpk(self):
+        if str(self.maxpk) != 'pk_is_natural':
+            self.maxpk = self.maxpk + 1
+            return self.maxpk
+        else:
+            raise TypeError
+
+    def __str__(self):
+        return u"occur: %d\nmaxpk: %d\n" % (self.occur, self.maxpk)
+
+
+def inspect(base):
+    models = {}
+
+    for item in base:
+        if not 'model' in item:
+            sys.stderr.write("malformed item, 'model' key doesn't exists: %s" % str(item))
+            return 1
+        if not 'pk' in item:
+            sys.stderr.write("malformed item, 'pk' key doesn't exists" % str(item))
+            return 1
+        model = item['model']
+
+        md = models_descr[model]
+
+        if not model in models:
+            models[model] = ItemsInfo(item, pk_natural = md.pk_natural)
```

Comment:
PEP8 violation, `pk_natural = md.pk_natural` should be written

`pk_natural=md.pk_natural` (no spaces)